### PR TITLE
Optimized a ton of stuff

### PR DESCRIPTION
**Pull request by Mooy1**
Optimized a bunch of stuff, mainly SlimefunItem#getByItem which does a map lookup each item, it should only be used once per ItemStack. Currently untested btw.

### CODE REVIEW

File: src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java [hunk 1]
```diff
@@ -40,63 +42,52 @@
...
+                seconds -= ((Bee) bee1).getSpeedMultipler() * b1.getAmount();
+            }
+        }
+        
+        ItemStack b2 = inv.getItemInSlot(getInputSlots()[1]);
+        
+        if (b2 != null) {
+            SlimefunItem bee2 = SlimefunItem.getByItem(b2);
+            if (bee2 instanceof Bee) {
+                seconds -= ((Bee) bee2).getSpeedMultipler() * b2.getAmount();
```
Comment (ProfElements) [inline src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java:86]: This doesn't factor in the change of time when 2 stacks of same bee type are in the MaterialHive.

File: src/main/java/me/profelements/dynatech/items/misc/DimensionalHomeDimension.java [hunk 2]
```diff
@@ -32,24 +35,4 @@ public ChunkData generateChunkData(World world, Random random, int chunkx, int c
...
-    @Override
-    public boolean shouldGenerateStructures() {
-        return false;
-    }
-
-    @Override
-    public boolean shouldGenerateMobs() {
-        return false;
-    }
-
```
Comment (ProfElements) [inline src/main/java/me/profelements/dynatech/items/misc/DimensionalHomeDimension.java:54]: Why were these removed? None of that should happen when generating the DimensonalHomeDimension

**Review by ProfElements — CHANGES_REQUESTED** [review]

File: src/main/java/me/profelements/dynatech/items/misc/DimensionalHomeDimension.java (hunk 2 above)
Comment (Mooy1) [reply src/main/java/me/profelements/dynatech/items/misc/DimensionalHomeDimension.java:54]: those are the default values

File: src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java (hunk 1 above)
Comment (Mooy1) [reply src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java:86]: ah ill fix that. How much of a bonus is it supposed to be?

Comment (ProfElements) [reply src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java:86]: Regular bees are 1500, Robotic Bees are 900, Advanced Robotic Bees are 600

File: src/main/java/me/profelements/dynatech/items/misc/DimensionalHomeDimension.java (hunk 2 above)
Comment (ProfElements) [reply src/main/java/me/profelements/dynatech/items/misc/DimensionalHomeDimension.java:54]: Ohhhhh I didnt know that.

File: src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java (hunk 1 above)
Comment (Mooy1) [reply src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java:86]: so if you have 2 regular, it should be 1500 - the speed boosts * stack amounts correct?

Comment (Mooy1) [reply src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java:86]: o i see its only when both are at max stack size aswell

Comment (ProfElements) [reply src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java:86]: yep

Comment (ProfElements) [reply src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java:86]: > so if you have 2 regular, it should be 1500 - the speed boosts * stack amounts correct?

so if you have 2 regular its just 1800 - speed amounts

if you have 2 stacks of regular it just makes the time = 1500

Comment (Mooy1) [reply src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java]: i changed it to just subtracting and extra 32 bees worth, because what you had actually made it worse when you had 2 stacks of robotic

File: src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java [hunk 3]
```diff
@@ -70,20 +66,34 @@ public MachineRecipe findNextRecipe(BlockMenu inv) {
...
+            if (sfItem instanceof Bee) {
+                Bee bee2 = (Bee) sfItem;
+
+                // subtract time
+                seconds -= bee2.getSpeedMultipler() * b2.getAmount();
+                
+                // if same type and both max stack size, add 32 bees worth of boost
+                if (bee1 == bee2 && b1.getAmount() == 64 && b2.getAmount() == 64) {
+                    seconds -= bee1.getSpeedMultipler() * 32;
+                }
```
Comment (ProfElements) [inline src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java]: Why are you changing how the MaterialHive works?

File: src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java (hunk 1 above)
Comment (ProfElements) [reply src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java]: How does all the bees differ then?

**Comment by ProfElements** [comment]
Issue with ItemBands using your optimizations

File: src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java (hunk 1 above)
Comment (Mooy1) [reply src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java]: I changed some of the base values of each bee, so with 2 stacks of same type its:
normal 1480
robotic 840
advanced robotic 520

**Comment by ProfElements** [comment]
Thank you for the pr! 👍  Just merged.
